plugins/modules/ec2_instance.py

```diff
@@ -1332,20 +1333,48 @@ def value_wrapper(v):
     param_mappings = [
         ParamMapper('ebs_optimized', 'EbsOptimized', 'ebsOptimized', value_wrapper),
         ParamMapper('termination_protection', 'DisableApiTermination', 'disableApiTermination', value_wrapper),
+        ParamMapper('security_groups', 'Groups', 'groupSet', value_wrapper),
+        ParamMapper('security_group', 'Groups', 'groupSet', value_wrapper),
         # user data is an immutable property
         # ParamMapper('user_data', 'UserData', 'userData', value_wrapper),
     ]
 
     for mapping in param_mappings:
         if params.get(mapping.param_key) is not None and mapping.instance_key not in skip:
```

Comment:
This nested logic is starting to get a little confusing.  IMO this 'if' would be better handled as a set of positive statements (Boolean logic can also easily catch you out with the way negations work):

```
        if mapping.instance_key in skip:
            continue
        if params.get(mapping.param_key) is None:
            continue
```

We can then drop one level of nesting.